### DISCUSSION TITLE
Authorization cookie not working after being set on the browser

### DISCUSSION THREAD

**Discussion by C4rmine** [cockpit-project/cockpit#18482] [Q&A]
### Explain what happens

1. Login throught requests and grab the authorization cookie
2. Set the authorization cookie in the browser using javascript statement "document.cookie"
3. Get response 401 unauthorized also if i put the creedentials in headers

### Version of Cockpit

215

### Where is the problem in Cockpit?

Applications

### Server operating system

Ubuntu

### What browsers are you using?

Chrome

**Comment by martinpitt** [comment]
Sorry for the late reply. I'm afraid there is not enough detail here to reproduce this. What do you mean with "using javascript statement `document.cookie`? You cannot set cookies that way in a persistent manner, you have to actually put them into the browser's cookie database. This generally works fine, you can e.g. get a cookie with curl

```
$ curl -kI -u admin:foobar --cookie-jar /tmp/c https://127.0.0.2:9091/cockpit/login
HTTP/1.1 200 OK
WWW-Authenticate: *
Content-Type: application/json
Set-Cookie: cockpit=dj0yO2s9ODY1N2NlZTc2ZGFhZDcyNzE2YjI3NThjYWI1OWVmN2Y0NjVkMmFiNDZkN2FkNzVjNDAwNTA2NDY5ODRmODg4Yg==; Path=/; SameSite=Strict; Secure; HttpOnly
```
and then use that token:

```
$ curl -kI --cookie-jar /tmp/c  https://127.0.0.2:9091/
HTTP/1.1 200 OK
[...]
```

**Comment by martinpitt** [comment]
What are you actually trying to do?
